to-hex payloadprocessor

EN10 opened the pull request:
Useful script I used for Natas 19 Challenge:
https://youtu.be/aZ3ZfAhdqT8

psiinon commented:
Thanks for sharing this :)
Could you add some details in the script, maybe including a link to that video?
Once its merged people wont see this commit so a bit of context would be useful.

EN10 commented:
> Thanks for sharing this :)
> Could you add some details in the script, maybe including a link to that video?
> Once its merged people wont see this commit so a bit of context would be useful.

thanks, this is my first ever pull :)

EN10 commented:
I think I'm done with edits?

psiinon commented:
Looks good to me :)
Just one more thing, could you add a comment to the changelog: https://github.com/zaproxy/community-scripts/blob/master/CHANGELOG.md

thc202 commented on payloadprocessor/to_hex.js:
```diff
@@ -0,0 +1,16 @@
+ /**
+ * Converts a string payload to hex.
+ * 
+ * Created to add functinality found in Burp to solve Natas19
```
There's a typo in functionality.

EN10 replied:
oops

thc202 commented on payloadprocessor/to_hex.js:
```diff
@@ -0,0 +1,16 @@
+ *
+ */
+
+function toHex(payload) {
```
It needs to keep the previous name to work properly.

EN10 replied:
ok

EN10 commented:
fixed typo, function name and added to changelog

EN10 commented:
is there anything else required?

thc202 commented:
Could you squash the commits? (The first is using your email, fine if you want it public though.)

EN10 commented:
> Could you squash the commits? (The first is using your email, fine if you want it public though.)

how do I do that?

kingthorin commented:
Here you go:

- https://www.instructables.com/id/How-to-Rebase-Interactive-With-Eclipse-eGit/
- https://www.kernel.org/pub/software/scm/git/docs/git-rebase.html#_interactive_mode
- https://wiki.eclipse.org/EGit/User_Guide#Interactive_Rebase

EN10 commented:
> Here you go:
> 
> * https://www.instructables.com/id/How-to-Rebase-Interactive-With-Eclipse-eGit/
> * https://www.kernel.org/pub/software/scm/git/docs/git-rebase.html#_interactive_mode
> * https://wiki.eclipse.org/EGit/User_Guide#Interactive_Rebase

ok thanks, I'm using Bash rather than Eclipse.
So I used: git rebase --autostash
hope that works?

kingthorin commented:
The middle link has CLI details. These might also help you:
- https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history
- https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d

kingthorin commented:
If you're having trouble, let me know I can tackle it.

thc202 commented:
Some more resources:
 - https://help.github.com/en/github/using-git/setting-your-username-in-git
 - https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address
 - https://help.github.com/en/github/using-git/about-git-rebase

EN10 commented:
> If you're having trouble, let me know I can tackle it.

thanks, for the help.

i'm getting this issue:
```
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_ro_RO/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_ru_RU/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_si_LK/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_sk_SK/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_sl_SI/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_sq_AL/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_sr_CS/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_sr_SP/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_tr_TR/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_ur_PK/contents/images/script-share.png: needs update
src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_zh_CN/contents/images/script-share.png: needs update
You must edit all merge conflicts and then
mark them as resolved using git add
```
seems to be stopping me from rebase

thc202 commented:
You need to force push (`-f` flag).

EN10 commented:
> You need to force push (`-f` flag).

thanks for your patience, i've realised i have edited a bunch of images in my last commit and trying to undo that.

kingthorin commented:
Check `git status` to see what it thinks is pending.

EN10 commented:
> Check `git status` to see what it thinks is pending.

git status

    On branch master
    Your branch is up-to-date with 'origin/master'.
    Changes not staged for commit:
      (use "git add <file>..." to update what will be committed)
      (use "git checkout -- <file>..." to discard changes in working directory)
    modified:   src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_tr_TR/contents/images/script-share.png
    modified:   src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_ur_PK/contents/images/script-share.png
    modified:   src/main/javahelp/org/zaproxy/zap/extension/communityScripts/resources/help_zh_CN/contents/images/script-share.png

EN10 commented:
ok i think i've squashed the commits, and removed email.

thc202 commented:
The committer name/email is probably not correct.

EN10 commented:
> The committer name/email is probably not correct.

thanks, I used this to fix for reference.

https://stackoverflow.com/questions/3042437/how-to-change-the-commit-author-for-one-specific-commit

thc202 commented:
Thank you!
